dvc/remote/http.py

```diff
@@ -43,36 +27,36 @@ def __init__(self, repo, config):
         self.path_info = self.path_cls(url) if url else None
 
     def _download(self, from_info, to_file, name=None, no_progress_bar=False):
-        callback = None
-        if not no_progress_bar:
-            total = self._content_length(from_info.url)
-            if total:
-                callback = ProgressBarCallback(name, total)
-
         request = self._request("GET", from_info.url, stream=True)
-
-        with open(to_file, "wb") as fd:
-            transferred_bytes = 0
-
-            for chunk in request.iter_content(chunk_size=self.CHUNK_SIZE):
-                fd.write(chunk)
-                fd.flush()
-                transferred_bytes += len(chunk)
-
-                if callback:
-                    callback(transferred_bytes)
+        with Tqdm(
+            total=None if no_progress_bar else self._content_length(from_info),
```

Comment:
`self._content_length()` should be dropped here. It performs not needed HEAD request, it is also never used elsewhere.

Reply:
no, it's been changed so the head request will not be re-done.

Reply:
specifically `getattr(url_or_request, "headers", ...)` won't redo the `HEAD` request.